### PR TITLE
ci: Add SSE contract test support

### PR DESCRIPTION
**Pull request by eli-darkly**
See: https://github.com/launchdarkly/sse-contract-tests

An example of this running in CI is [here](https://app.circleci.com/pipelines/github/launchdarkly/ruby-eventsource/166/workflows/a476ac4f-9fc6-4f39-b4fc-e059ab1819bf/jobs/869). That workflow is currently failing due to problems that appear to be JRuby-specific (see below); in all of the non-JRuby jobs, everything passes and I believe the SSE implementation is now fully compliant with spec.

The two kinds of failures I'm seeing in JRuby (which I haven't yet filed separate stories for, because they're so odd that I need to rule out problems in the test logic first) appear to be problems in HTTP client behavior, not in SSE parsing:

* It seems to be unable to detect a dropped connection if it's in the middle of parsing an event. (Test: "reconnection/discards partial messages on retry")
* It sometimes stops receiving data from the stream if the data is being sent in 1-character chunks. In case there's something I don't understand about httprb, or a related known issue, I've filed https://github.com/httprb/http/issues/703 to seek insight.

If we can fix those, then we will have much better test coverage in JRuby than before, since as described in https://github.com/launchdarkly/ruby-eventsource/pull/27, the mechanism we've been using to do end-to-end HTTP tests within the _unit_ tests for this project was extremely unreliable in JRuby.

If you'd like to test this locally in any version of Ruby, clone this repo and also https://github.com/launchdarkly/sse-contract-tests. In ruby-eventsource, run `make build-contract-tests` and then `make start-contract-test-service`. Then, in sse-contract-tests, run `make` and then `./sse-contract-tests -url http://localhost:8000` - that's it.

### CODE REVIEW

File: contract-tests/service.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  end
+
+  streams[streamId] = entity
+
+  return [201, {"Location" => streamResourceUrl}, nil]
+end
+
+delete '/streams/:id' do |streamId|
+  entity = streams[streamId]
+  return 404 if stream.nil?
```
Comment (keelerm84) [inline contract-tests/service.rb:72]: Should this be ...?
```suggestion
  return 404 if entity.nil?
```

File: contract-tests/stream_entity.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'ld-eventsource'
+require 'json'
+require 'net/http'
+
+set :port, 8000
+set :logging, false
+
+streams = {}
+streamCounter = 0
```
Comment (keelerm84) [inline contract-tests/stream_entity.rb:9]: I don't think these two are needed anymore.

Comment (eli-darkly) [reply contract-tests/stream_entity.rb:9]: Indeed, that's a copy-paste mistake.

File: contract-tests/service.rb (hunk 1 above)
Comment (eli-darkly) [reply contract-tests/service.rb:72]: 🤦

**Review by keelerm84 — APPROVED** [review]

**Comment by keelerm84** [comment]
I have updated this PR so it runs on GH actions instead of circleci.

The problems Eli mentioned still seem to be present, so I've just disabled the contract tests in jruby for now. Might as well get some value from them now instead of waiting until the jruby stuff gets resolved (if it ever does).

**Review by cwaldren-ld — APPROVED** [review]

**Pull request merged**